core/src/main/java/de/danoeh/antennapod/core/service/playback/ExoPlayerWrapper.java

```diff
@@ -243,7 +243,8 @@ public void setDataSource(String s, String user, String password)
             );
             httpDataSourceFactory.setDefaultRequestProperties(requestProperties);
         }
-        DataSource.Factory dataSourceFactory = new DefaultDataSourceFactory(context, null, httpDataSourceFactory);
+        DataSource.Factory dataSourceFactory = new DefaultDataSource.Factory(context, httpDataSourceFactory)
+                .setTransferListener(null);
```

Comment:
What does this transfer listener do? Do we need to set it for the cached version as well (3 lines below)?

Reply:
It looks for data transfer events; however...

I just checked the source code for the old DefaultDataSourceFactory, and it looks like if you used `DefaultDataSourceFactory(context, httpDataSourceFactory)`, the TransferListener was null by default. It looks like we don't use it at all, then. We can remove it entirely.